src/main/java/io/github/railroad/platform/fabric/FabricMod.java

```diff
@@ -0,0 +1,376 @@
+package io.github.railroad.platform.fabric;
+
+import java.io.BufferedInputStream;
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.FileNotFoundException;
+import java.io.FileOutputStream;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.io.Reader;
+import java.net.URL;
+import java.nio.charset.Charset;
+import java.util.Properties;
+
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+
+/*
+ * @author lifebearrrr
+ * 
+ * Used to generate new Fabric Mod with all assets
+ * 
+ *TODO: Replace my Fabric version getter (that doesn't get the right fabric api version)
+ *with Saksham4106's Fabric version getter when it's done
+ */
+@SuppressWarnings("static-access")
+public class FabricMod {
+	
+	private static File project;
+	private static String pkg;
+	private static String name;
+	private static String modID;
+	private static String workspacePath;
+
+	public FabricMod(String name, String modId, String pkg, String version,
+			String workspace/* Put file.getAbsolutePath() or string in this format "C:\\Users\\name\\" */) {
+		this.workspacePath = workspace + "\\";
+		this.pkg = pkg;
+		this.name = name;
+		this.modID = modId;
+		generateMod(name, modId, pkg, version);
+	}
+
+	public static void generateMod(String name, String modID, String pkg, String version) {
+		File workspace = getOrCreateFolder(workspacePath);
+
+		File proj = createDirs(workspace, name, modID);
+		
+		project = proj;
+
+		downloadFiles(name, modID);
+
+		setUpFabric(workspace.getAbsolutePath(), name, modID, version, pkg);
+
+		setUpGradle(name, modID, pkg, version,
+				proj + "\\" + "gradle.properties", proj);
+	}
+
+	public static File createDirs(File workspace, String name, String modID) {
+		// Creating Folders / Files
+		File proj = getOrCreateFolder(workspace.getAbsolutePath() + "\\" + name);
+		getOrCreateFolder(proj + "\\" + "src\\main\\resources\\assets\\" + modID);
+		getOrCreateFolder(proj + "\\" + "src\\main\\resources\\data\\" + modID);
+		getOrCreateFile(proj + "\\" + "gradle.properties");
+		return proj;
+	}
+
+	public static void downloadFiles(String name, String modID) {

```

Comment:
Instead of downloading each individual file like that, why not just download the whole thing, and then delete the files we don't need?

Reply:
Downloading the entire zip file takes a surprisingly longer amount of time to download. I've compared what I'm downloading with what the entire thing looks like, and there aren't major differences, so I'm not sure why it takes so much longer to download.

Reply:
i mean, how much longer are we speaking?

Reply:
It seems it isn't just the download that takes a bit longer it's the combination of the download and unzipping. Which overall takes about 2 times longer to do.

Reply:
why can't you just clone the fabric example mod repo